### PR TITLE
[ws-scheduler] Prevent ghost explosions

### PR DESCRIPTION
**Pull request by csweichel**
ws-scheduler's driver had a state management issue where it relied on Kubernetes to update its state quickly enough. If that didn't happen, we'd start way too many ghosts.

This change maintains a more up-to-date local state to prevent this issue.

### CODE REVIEW

File: components/ee/ws-scheduler/pkg/scaler/driver.go [hunk 1]
```diff
@@ -151,6 +151,18 @@ func (wspd *WorkspaceManagerPrescaleDriver) Run() {
...
+	canStartGhost := func() bool {
+		if status.Count.Ghost >= wspd.Config.MaxGhostWorkspaces {
+			log.WithField("limit", wspd.Config.MaxGhostWorkspaces).Warn("max number of ghost workspace reached")
+			return false
+		}
+		return true
+	}
+	registerStartingGhost := func(id string) {
+		startingGhosts[id] = wspd.time.Now()
+		status.Count.Ghost++
```
Comment (geropl) [inline components/ee/ws-scheduler/pkg/scaler/driver.go:163]: I'm just realizing we re counting ghost twice now: IMO [L163](https://github.com/gitpod-io/gitpod/pull/3357/files#diff-4678ff698bf4d02fdd0f9423962d464d8f12c119a7f297987962f89945216769L163) should be removed.
@csweichel WDYS?

Comment (csweichel) [reply components/ee/ws-scheduler/pkg/scaler/driver.go:163]: where else do we count the ghost? Note that we never use `len(startingGhosts)`

Comment (geropl) [reply components/ee/ws-scheduler/pkg/scaler/driver.go:163]: We do, in [L163](https://github.com/gitpod-io/gitpod/pull/3357/files#diff-4678ff698bf4d02fdd0f9423962d464d8f12c119a7f297987962f89945216769L163)

Comment (geropl) [reply components/ee/ws-scheduler/pkg/scaler/driver.go:163]: `status.Count.Ghost += len(startingGhosts)`

Comment (csweichel) [reply components/ee/ws-scheduler/pkg/scaler/driver.go:163]: Thanks - now I've seen it :) 

Fixed

**Review by geropl — APPROVED** [review]
:+1:

**Pull request merged**